Celeste.Mod.mm/Mod/UI/OuiModUpdateList.cs

```diff
@@ -119,18 +134,29 @@ class OuiModUpdateList : Oui, OuiModOptions.ISubmenu {
                                 versionUpdate = $"{metadata.VersionString} > {update.Version}";
 
                             TextMenu.Button button = new TextMenu.Button($"{metadata.Name.SpacedPascalCase()} | v. {versionUpdate} ({new DateTime(1970, 1, 1, 0, 0, 0, 0).AddSeconds(update.LastUpdate):yyyy-MM-dd})");
+                            modUpdateHolder updateAllData = new modUpdateHolder() { update = update, metadata = metadata, button = button };
                             button.Pressed(() => {
                                 // make the menu non-interactive
                                 menu.Focused = false;
                                 button.Disabled = true;
 
                                 // trigger the update download
                                 downloadModUpdate(update, metadata, button);
+
+                                // remove this mod from the "update all" list and, if necessary, disable the "update all" button
+                                updateableMods.Remove(updateAllData);
+                                if (updateableMods.Count == 0) {
+                                    updateAllButton.Disabled = true;
+                                }
```

Comment:
I've moved it to `downloadAllMods` because I need the reference to the `ModUpdateHolder` instance corresponding to the mod to be updated, wich I do not have in `doDownloadModUpdate`